src/Compilers/CSharp/Test/Semantic/Semantics/AnonymousFunctionTests.cs

```diff
@@ -0,0 +1,802 @@
+            var source = @"
+using System;
+
+public class C
+{
+    public void F()
+    {
+        Func<int> f = () =>
+        {
+            int i = 0;
+            Func<int> g = static () => i;
+            return 0;
+        };
+    }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (11,27): error CS8652: The feature 'static anonymous function' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //             Func<int> g = static () => i;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "static").WithArguments("static anonymous function").WithLocation(11, 27),
+                // (11,40): error CS8427: A static anonymous function cannot contain a reference to 'i'.
+                //             Func<int> g = static () => i;
+                Diagnostic(ErrorCode.ERR_StaticAnonymousFunctionCannotCaptureVariable, "i").WithArguments("i").WithLocation(11, 40));
+        }
+
+        [Fact]
+        public void StaticLambdaCanHaveLocalsCapturedByInnerInstanceLocalFunction()
+        {
+            var source = @"
+using System;
+
+public class C
+{
+    public void F()
+    {
+        Func<int> f = static () =>
+        {
+            int i = 0;
+            int g() => i;
+            return g();
+        };
+    }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,23): error CS8652: The feature 'static anonymous function' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         Func<int> f = static () =>
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "static").WithArguments("static anonymous function").WithLocation(8, 23));
+        }
+
+        [Fact]
+        public void StaticLambdaCannotHaveLocalsCapturedByInnerStaticLocalFunction()
+        {
+            var source = @"
+using System;
+
+public class C
+{
+    public void F()
+    {
+        Func<int> f = static () =>
+        {
+            int i = 0;
+            static int g() => i;
+            return g();
+        };
+    }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,23): error CS8652: The feature 'static anonymous function' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         Func<int> f = static () =>
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "static").WithArguments("static anonymous function").WithLocation(8, 23),
+                // (11,31): error CS8421: A static local function cannot contain a reference to 'i'.
+                //             static int g() => i;
+                Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureVariable, "i").WithArguments("i").WithLocation(11, 31));
+        }
+
+        [Fact]
+        public void InstanceLambdaCannotHaveLocalsCapturedByInnerStaticLocalFunction()
```

Comment:
>InstanceLambdaCannotHaveLocalsCapturedByInnerStaticLocalFunction [](start = 20, length = 64)

The static local function cases should be covered in Microsoft.CodeAnalysis.CSharp.UnitTests.LocalFunctionTests.

Reply:
I'm fine with a little split over these files. This was specifically trying to to validate that the outer instance lambda doesn't change anything.